### PR TITLE
[IMP] Add env var expansion for repo.yml

### PR DESCRIPTION
**Pull request by lasley**
* Expand environment variables in repo.yml before use

My use case being a dynamic odoo version, which interestingly enough could also work for module branches too.

### CODE REVIEW

File: build.d/20-repos-aggregate [hunk 1]
```diff
@@ -11,6 +11,7 @@ conf=/opt/odoo/custom/src/repos.yaml
 if [ -f $conf ]; then
     log INFO Aggregating repositories from $conf
     cd $(dirname $conf)
+    echo $(eval echo $(cat $conf)) > $conf
```
Comment (Yajo) [inline build.d/20-repos-aggregate]: Better use envsubst, like with conf files.

**Comment by lasley** [comment]
> Don't you think PR for gitaggregator would be better?

Env vars seem a bit more on the side of the system configuration vs. a CLI tool IMO, but I could be wrong. TBH this is the first I've heard of the tool, so I'm not sure of a use case outside of this one. If you think I should submit that way though, I don't have any problems with it.

TIL with envsubst - not sure how I've gone this long in life without needing it! Updated.

**Comment by Yajo** [comment]
I added a little fix, but there's still a problem: this would break [the scaffolding's setup-devel step](https://github.com/Tecnativa/docker-odoo-base#development).

We could either support this also in [tecnativa/git-aggregator](https://hub.docker.com/r/tecnativa/git-aggregator) or you could propose this PR to [upstream](https://github.com/acsone/git-aggregator).

**Comment by lasley** [comment]
Hrmmm alright so out of the choice of git-aggregator Docker container vs. library, library is probably best. Seems like the feature would get buried. PR in https://github.com/acsone/git-aggregator/pull/8

**Comment by Yajo** [comment]
Nice, I merged acsone/git-aggregator#8 in our fork's master branch, so you have it in https://hub.docker.com/r/tecnativa/odoo-base/~/settings/automated-builds/

**Comment by lasley** [comment]
Hooray thanks @Yajo! This feature should help me quite a bit 😄

**Pull request closed without merging**